compute/cker/src/train/BinaryArithmetic.test.cc

```diff
@@ -117,14 +132,102 @@ TEST(CKer_Operation, SubGrad)
+      nnfw::cker::Shape{2, 3}, incoming_backward.data(), nnfw::cker::Shape{2, 3},
+      lhs_backward.data(), nnfw::cker::Shape{2, 3}, rhs_backward.data(),
+      nnfw::cker::train::ArithmeticType::kMul);
+
+    for (size_t i = 0; i < lhs_backward.size(); ++i)
+      EXPECT_FLOAT_EQ(lhs_backward[i], expected_lhs_backward[i]);
+    for (size_t i = 0; i < rhs_backward.size(); ++i)
+      EXPECT_FLOAT_EQ(rhs_backward[i], expected_rhs_backward[i]);
+  }
+  {
+    // Shape: {4, 3}
+    std::vector<float> lhs = {-14, 15, -16, 17, -18, 19, -20, 21, -22, 23, -24, 25};
+    std::vector<float> rhs = {-26, 27, -28, 29, -30, 31, -32, 33, -34, 35, -36, 37};
+    std::vector<float> incoming_backward = {-2, 3, -4, 5, -6, 7, -8, 9, -10, 11, -12, 13};
+    std::vector<float> expected_lhs_backward(12);
+    std::vector<float> expected_rhs_backward(12);
+    std::vector<float> lhs_backward(12);
+    std::vector<float> rhs_backward(12);
+
+    for (uint32_t i = 0; i < 12; ++i)
+    {
+      expected_lhs_backward[i] = incoming_backward[i] * rhs[i];
+      expected_rhs_backward[i] = incoming_backward[i] * lhs[i];
+    }
+
+    nnfw::cker::train::BinaryArithmeticGrad(
+      nnfw::cker::Shape{4, 3}, lhs.data(), nnfw::cker::Shape{4, 3}, rhs.data(),
+      nnfw::cker::Shape{4, 3}, incoming_backward.data(), nnfw::cker::Shape{4, 3},
+      lhs_backward.data(), nnfw::cker::Shape{4, 3}, rhs_backward.data(),
+      nnfw::cker::train::ArithmeticType::kMul);
+
+    for (size_t i = 0; i < lhs_backward.size(); ++i)
+      EXPECT_FLOAT_EQ(lhs_backward[i], expected_lhs_backward[i]);
+    for (size_t i = 0; i < rhs_backward.size(); ++i)
+      EXPECT_FLOAT_EQ(rhs_backward[i], expected_rhs_backward[i]);
+  }
+  {
+    // Shape: {2, 2, 2}
+    std::vector<float> lhs = {-10, 11, -12, 13, -14, 15, -16, 17};
+    std::vector<float> rhs = {-18, 19, -20, 21, -22, 23, -24, 25};
+    std::vector<float> incoming_backward = {-2, 3, -4, 5, -6, 7, -8, 9};
+    std::vector<float> expected_lhs_backward = incoming_backward;
+    std::vector<float> expected_rhs_backward = incoming_backward;
+    std::vector<float> lhs_backward(8);
+    std::vector<float> rhs_backward(8);
+
+    for (uint32_t i = 0; i < 8; ++i)
+    {
+      expected_lhs_backward[i] = incoming_backward[i] * rhs[i];
+      expected_rhs_backward[i] = incoming_backward[i] * lhs[i];
+    }
+
+    nnfw::cker::train::BinaryArithmeticGrad(
+      nnfw::cker::Shape{2, 2, 2}, lhs.data(), nnfw::cker::Shape{2, 2, 2}, rhs.data(),
+      nnfw::cker::Shape{2, 2, 2}, incoming_backward.data(), nnfw::cker::Shape{2, 2, 2},
+      lhs_backward.data(), nnfw::cker::Shape{2, 2, 2}, rhs_backward.data(),
+      nnfw::cker::train::ArithmeticType::kMul);
+
+    for (size_t i = 0; i < lhs_backward.size(); ++i)
+      EXPECT_FLOAT_EQ(lhs_backward[i], expected_lhs_backward[i]);
+    for (size_t i = 0; i < rhs_backward.size(); ++i)
+      EXPECT_FLOAT_EQ(rhs_backward[i], expected_rhs_backward[i]);
+  }
+}
+
```

Comment:
As I remember, we need to add the equal number of negative tests when tests are added. We may need a negative test. It may be enough to add different shape of `lhs` and `rhs`.